aibolit/patterns/string_concat/string_concat.py

```diff
@@ -1,54 +1,50 @@
+    '''
+    Any usage string concatenation using '+' operator is considered as a pattern.
+    '''
+    def value(self, filename: str) -> List[int]:
+        lines: Set[int] = set()
+        ast = AST.build_from_javalang(build_ast(filename))
+        for node in ast.get_proxy_nodes(ASTNodeType.BINARY_OPERATION):
             if node.operator == '+':
-                is_l_literal = isinstance(node.operandl, javalang.tree.Literal)
-                is_r_literal = isinstance(node.operandr, javalang.tree.Literal)
-                is_r_member = isinstance(node.operandr, javalang.tree.MemberReference)
-                is_l_member = isinstance(node.operandl, javalang.tree.MemberReference)
-                is_l_meth_inv = isinstance(node.operandl, javalang.tree.MethodInvocation)
-                is_r_meth_inv = isinstance(node.operandr, javalang.tree.MethodInvocation)
-                is_l_this = isinstance(node.operandl, javalang.tree.This)
-                is_r_this = isinstance(node.operandr, javalang.tree.This)
-                if is_l_literal and (is_r_member or is_r_meth_inv or is_r_this):
-                    is_string_literal = '"' in node.operandl.value  # type: ignore
-                    if is_string_literal:
-                        if node.operandl.position:
-                            lines.add(node.operandl.position.line)
-                        elif node.operandr.position:
-                            lines.add(node.operandr.position.line)
-                        elif hasattr(node.operandl, '_position'):
-                            lines.add(node.operandl._position.line)
-                        elif hasattr(node.operandr, '_position'):
-                            lines.add(node.operandr._position.line)
-                elif is_r_literal and (is_l_member or is_l_meth_inv or is_l_this):
-                    is_string_literal = '"' in node.operandr.value  # type: ignore
-                    if is_string_literal:
-                        if node.operandl.position:
-                            lines.add(node.operandl.position.line)
-                        elif node.operandr.position:
-                            lines.add(node.operandr.position.line)
-                        elif hasattr(node.operandr, '_position'):
-                            lines.add(node.operandr._position.line)
-                        elif hasattr(node.operandl, '_position'):
-                            lines.add(node.operandl._position.line)
+                if hasattr(node.operandl, 'value') and isinstance(node.operandl.value, str) and \
+                        not node.operandl.value.isdigit():
+                    if node.operandl.line:
+                        lines.add(node.operandl.line)
+                    elif node.operandr.line:
+                        lines.add(node.operandr.line)
+
+                elif hasattr(node.operandr, 'value') and isinstance(node.operandr.value, str) and \
```

Comment:
It sounds like we check only if one argument of `BINARY_OPERATION` is string, while comment in class says we are looking for both arguments to be string. If that correct, we have to check both operands to be string.

Reply:
we search  for any string concatenation. but there can be different concatenation:
`"String" + "4"`
`int a = "b";`
`a+ "b";`
`b + "a";`
`"; " + message.toString()`
```

            Integer d = 1 + b;
            System.out.println(a + b);
```
etc.

look at the tests, there are examples there.

Reply:
So it just look for using concatenation with any string literal?
If so, we better create a single function to check for a string literal and use it to check left and right side of operators.

Reply:
ok

Reply:
@Vitaly-Protasov please move logic of checking operand to separate function and call it in `operandr` and `operandl` here.